### PR TITLE
[entropy_src/rtl] make sha3_rst_storage_err fatal

### LINKED ISSUE #14183: [entropy_src/sha3] Hook-up reset storage error
sha3 has added a security feature #14103 to ensure that storage resets/clears cannot be easily attacked. If attacked, this would result in the storage being all 0, which is very undesirable for the sha3 condition use case.

This issue is filed to make sure we hook-up the error indication as a proper alert.

### PR DESCRIPTION
**Pull request by mwbranstad**
Connecting the sha3 output cakk sha3_rst_storage_err as another
source for fatal errors, similar to sha3_count_error.
Fixes #14183.

Signed-off-by: Mark Branstad <mark.branstad@wdc.com>

### CODE REVIEW

File: hw/ip/entropy_src/rtl/entropy_src_core.sv [hunk 1]
```diff
@@ -416,12 +416,13 @@ module entropy_src_core import entropy_src_pkg::*; #(
...
   logic                    es_cntr_err_sum;
   logic                    sha3_state_error_sum;
+  logic                    sha3_rst_storage_err_sum;
   logic                    efuse_es_sw_reg_en;
   logic                    efuse_es_sw_ov_en;
 
   logic                    sha3_state_error;
   logic                    sha3_count_error;
-  logic                    sha3_rst_storage_error;
+  logic                    sha3_rst_storage_err;
```
Comment (martin-lueker) [inline hw/ip/entropy_src/rtl/entropy_src_core.sv:425]: Out of curiousity, why the change of name here?

**Review by martin-lueker — APPROVED** [review]
LGTM

File: hw/ip/entropy_src/rtl/entropy_src_core.sv (hunk 1 above)
Comment (mwbranstad) [reply hw/ip/entropy_src/rtl/entropy_src_core.sv:425]: Long standing habit that no design signal or file or anything should have the word "error" in it, since this should be considered reserved. Simulation post-processors need to search for "error" and this is commonly used by various IP blocks, both for VIP and technology books. So better to avoid using it in the design domain.

**Review by eunchan — APPROVED** [review]
